src/core/lib/security/credentials/tls/grpc_tls_certificate_provider.cc

```diff
@@ -364,6 +365,54 @@ FileWatcherCertificateProvider::ReadIdentityKeyCertPairFromFiles(
   return absl::nullopt;
 }
 
+absl::StatusOr<bool> PrivateKeyAndCertificateMatch(
+    absl::string_view private_key, absl::string_view cert) {
+  if (private_key.empty()) {
+    return absl::InvalidArgumentError("Private key string is empty.");
```

Comment:
How could `absl::InvalidArgumentError` be of type `absl::StatusOr<bool>`?

Reply:
`absl::InvalidArgumentError()` returns an `absl::Status` for which `absl::StatusOr<T>` has a constructor.